src/main/java/models/League.java

```diff
@@ -0,0 +1,46 @@
+package models;
+
+import java.util.LinkedList;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.Queue;
+
+public class League {
+
+    private Queue<Team> initialTeams;
+    private HashMap<Integer, ArrayList<Team>> winners;
```

Comment:
Consider using something like a `Pair` here rather than an `ArrayList`, since there are always 2 teams in a match